### DISCUSSION TITLE
What would you want of the eraser tool?

### DISCUSSION THREAD

**Discussion by bhennion** [xournalpp/xournalpp#2767] [General]
Hi everyone,

Working on #2729, I've come to work on the eraser tool. I've noticed some open issues #1339 #997 on the matter. Because adapting the eraser to splines takes some efforts, I'd better get it right the first time. So before I waste to much time: what do you expect of the eraser tool?

Here are things I gathered from the above open issues

- Allow for various shapes (square, circle)
- Be pressure sensitive
- Fix #997 (eraser does not work well with highlighter)
- In standard and whiteout mode, the eraser should erase exactly what's under its cursor. No more, no less. (It ain't the case right now)

It there any other eraser-related bug I missed? Any enhancement someone would like to add?

**Comment by rolandlo** [comment]
The eraser is very slow sometimes. Just create a long winded stroke like this one:
[slow1.zip](https://github.com/xournalpp/xournalpp/files/5939759/slow1.zip)
Try to erase where I ended the stroke (near the center). You can create more extreme examples in this way where you have to wait for rather long time until things get erased.

**Reply by bhennion** [reply]
I guess it's from copying the entire stroke into a list. It can probably be avoided. Thanks for pointing it out!
